### ISSUE TITLE
Bug: Assignment to an input function parameter - C backend

### ISSUE THREAD

**Issue by ronnuriel** [lcompilers/lpython#1952]
semantic error: Assignment to an input function parameter `x` is not allowed
 --> test:2:5
  |
2 |     x = u16(11)
  |   

```
def foo(x: u16):
    x = u16(11)

foo(u16(10))
```

**Comment by Shaikh-Ubaid** [comment]
>semantic error: Simple Type u16 cannot be intent InOut/Out

This is expected. Simple types cannot be `InOut`. In `CPython` if a variable (of any Simple Type `i32`, `i64`, `u32`, `f32`, ...) is updated in the callee function then the updated value is not reflected back in the caller function. This happens because `CPython` passes arguments by (copy of) object reference. When an object is updated inside a callee function, the original reference/variable remains intact, I think.

**Comment by Shaikh-Ubaid** [comment]
```bash
$ cat examples/expr2.py 
def foo(x: u16):
    x = u16(11)

foo(u16(10))
$ lpython examples/expr2.py
semantic error: Assignment to an input function parameter `x` is not allowed
 --> examples/expr2.py:2:5
  |
2 |     x = u16(11)
  |     ^ Use InOut[x] to allow assignment


Note: if any of the above error or warning messages are not clear or are lacking
context please report it to us (we consider that a bug that must be fixed).
```
It seems the bug here is the guiding message to use `InOut`. In this case, using `InOut` for a `Simple Type` is not allowed.

**Comment by certik** [comment]
Yes, for simple types we need to improve the error message not to suggest InOut.

@ronnuriel are you trying to return a value from the function using the argument, or are you trying to just create a new local variable that happens to have the same name as an argument?

For now we decided not to allow "shadowing" arguments, to prevent bugs. If you don't need the argument, just rename it (or rename the local variable).

**Comment by ronnuriel** [comment]
@certik No, Eyal tried to cast the var but he couldn't..
